### PR TITLE
chore: Transaction Component Add Paymaster Support + QA cleanup

### PR DESCRIPTION
**Pull request by ilikesymmetry**
**What changed? Why?**

- Add `SmartWalletCapabilities` paymaster sponsorship support in the `Transaction Component`. 
- Add new `click` contract for testing.
- Remove `AppBaseSepolia`. 
- Add `chainId` parameter to the `Transaction Component`.
- Fix the bug where we would see `User rejected the request` error despite the transaction succeeding. 
- Auto switch the users wallet to Base Sepolia when interacting with the transaction component.
- Fix bug where successful testnet transaction's "View Transaction" button links to Base Mainnet explorer.
- General clean up.

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/transaction/components/Transaction.tsx [hunk 1]
```diff
@@ -7,11 +7,15 @@ export function Transaction({
   className,
   children,
   contracts,
+  chainId,
+  capabilities,
```
Comment (Zizzamia) [inline src/transaction/components/Transaction.tsx:11]: Keep alphabetic order. 

Btw in other places we have `chain` and not `chainId`, also why this should override the chain set at OnchainProvider level?

File: src/transaction/hooks/useWriteContracts.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import type { TransactionExecutionError } from 'viem';
 import { useWriteContracts as useWriteContractsWagmi } from 'wagmi/experimental';
+import { METHOD_NOT_SUPPORTED_ERROR_SUBSTRING } from '../constants';
```
Comment (Zizzamia) [inline src/transaction/hooks/useWriteContracts.ts:3]: Thank you for starting a `constants` file.

File: src/transaction/components/Transaction.tsx (hunk 1 above)
Comment (cpcramer) [reply src/transaction/components/Transaction.tsx:11]: Updated to alphabetic

File: src/transaction/components/Transaction.tsx [hunk 3]
```diff
@@ -12,6 +14,8 @@ export function Transaction({
   return (
     <TransactionProvider
       address={address}
+      capabilities={capabilities}
+      chainId={chainId}
```
Comment (Zizzamia) [inline src/transaction/components/Transaction.tsx:18]: To stay in line with the other components API, this should be
```ts
chain?: Chain;
```

File: src/transaction/components/Transaction.tsx [hunk 4]
```diff
@@ -4,6 +4,8 @@ import { TransactionProvider } from './TransactionProvider';
 
 export function Transaction({
   address,
+  capabilities,
+  chainId,
```
Comment (Zizzamia) [inline src/transaction/components/Transaction.tsx:8]: To stay in line with the other components API, this should be
```ts
chain?: Chain;
```

File: src/transaction/constants.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const SEND_CALLS_NOT_SUPPORTED_ERROR = 'SEND_CALLS_NOT_SUPPORTED_ERROR';
+export const METHOD_NOT_SUPPORTED_ERROR_SUBSTRING =
+  'this request method is not supported'; // most likely EOA
```
Comment (Zizzamia) [inline src/transaction/constants.ts:3]: If this message is displayed, we probably want the `t` upper case. as `This request`

File: src/transaction/components/TransactionProvider.tsx [hunk 6]
```diff
@@ -69,26 +69,54 @@ export function TransactionProvider({
     }
   }, [contracts, writeContract]);
 
+  const switchChain = useCallback(
```
Comment (Zizzamia) [inline src/transaction/components/TransactionProvider.tsx:72]: To keep this code easier to mantain and test, we should probably split those into indipendent custom Hooks.

Let me know how much this is douable or not.

File: site/docs/pages/transaction/transaction.mdx [hunk 7]
```diff
@@ -77,22 +78,31 @@ export default function TransactionComponents() {
...
+            address={address}
+            capabilities={{
+              paymasterService: {
+                url: 'https://api.developer.coinbase.com/rpc/v1/base-sepolia/KkgONfLFgUZKI8nnrttLbI78U2vyhtXg',
+              },
+            }}
+            chainId={84532}
+            contracts={contracts}
+            onError={onError}
+          >
```
Comment (ilikesymmetry) [inline site/docs/pages/transaction/transaction.mdx:104]: @fakepixels sample use of new `Transaction` component

**Review by Zizzamia — APPROVED** [review]
Approving this as it is, and than do a bunch of fast-follow to correct what's breaking.

**Pull request merged**